be/src/olap/row_block2.cpp

```diff
@@ -329,7 +329,8 @@ Status RowBlockV2::_copy_data_to_column(int cid, doris::vectorized::MutableColum
 }
 
 Status RowBlockV2::convert_to_vec_block(vectorized::Block* block) {
-    for (int i = 0; i < _schema.column_ids().size(); ++i) {
+    DCHECK_LE(block->columns(), _schema.column_ids().size());
+    for (int i = 0; i < block->columns(); ++i) {
```

Comment:
Maybe using ```schema.num_column_ids()```  instead of  ```block->columns()``` is better?

Reply:
vec block column size is different with RowBlockV2 size, the mismatch cause the problem